lib/concurrent/struct/immutable_struct.rb

```diff
@@ -0,0 +1,87 @@
+require 'concurrent/struct/abstract_struct'
+require 'concurrent/synchronization'
+
+module Concurrent
+
+  # A thread-safe, immutable variation of Ruby's standard `Struct`.
+  #
+  # @see http://ruby-doc.org/core-2.2.0/Struct.html Ruby standard library `Struct`
+  module ImmutableStruct
+    include AbstractStruct
+
+    # @!visibility private
+    def initialize(*values)
+      ns_initialize(*values)
+    end
```

Comment:
Currently `ImmutableStruct` doesn't extend `Synchronization::Object` since it's immutable (and the `#synchronize` call in the initializer creates a significant performance penalty). The only reason it has an `ns_initialize` method is to be consistent with the other struct classes. I could add `ensure_ivar_visibility!` here but I would need to add a class method `ensure_ivar_visibility!` to the synchronization classes. That's easy enough to do if necessary.